products/sle/main.pm

```diff
@@ -199,6 +199,39 @@ if (get_var('DEV_IMAGE')) {
     }
 }
 
+#This setting is used to enable all extensions/modules for upgrading tests, generate corresponding extensions/modules list for all addons, all extensions, all modules cases.
+if (check_var('SCC_REGISTER', 'installation') && get_var('ALL_ADDONS')) {
+    my $version = get_required_var("HDDVERSION");
+    #Below $common* store the extensions/modules list in common.
+    my $common_addons_12SP3     = 'sdk,pcm,tcm,wsm';
+    my $common_modules_12SP3    = 'pcm,tcm,wsm';
+    my $common_extensions_12SP3 = 'sdk';
+    my $common_addons_11SP4     = 'ha,geo,sdk';
+    #Below $external* store external extensions/modules list on each ARCH.
+    my %external_addons_12SP3
+      = ('x86_64', 'ha,geo,we,live,asmm,contm,lgm,hpcm', 'ppc64le', 'ha,live,asmm,contm,lgm', 's390x', 'ha,geo,asmm,contm,lgm', 'aarch64', 'hpcm');
```

Comment:
WDYT about rewriting it so it's on multiple lines and it's easy to see the mapping?

Reply:
Better to write hashes with the "big comma" operator, e.g.

```perl
my %external_addons_12SP3 = (
    x86_64 => 'ha,geo…',
    ppc64le => 'ha,live,asmm…',
…
);
```